utils/load_path.ml

```diff
@@ -94,13 +85,12 @@ let remove_dir dir =
    order to enforce left-to-right precedence. *)
 let add dir =
   assert (not Config.merlin || Local_store.is_bound ());
-  let new_files, new_files_uncap =
-    add_to_maps (Filename.concat dir.Dir.path) dir.Dir.files
-      SMap.empty SMap.empty
-  in
-  let first _ fn _ = Some fn in
-  files := SMap.union first !files new_files;
-  files_uncap := SMap.union first !files_uncap new_files_uncap;
+  List.iter (fun base ->
+    let fn = Filename.concat dir.Dir.path base in
+    if not (STbl.mem !files base) then begin
+      STbl.replace !files base fn;
+      STbl.replace !files_uncap (String.uncapitalize_ascii base) fn
+    end) dir.Dir.files;
```

Comment:
If I understand correctly, the behavior here is slightly different. The new behavior maybe be more correct (?), but it is worth discussing. Before, we would independently compute the union of `files` and of `files_uncap`, each time with the "oldest element wins" strategy.
Unless I am missing something, it is possible that the `files_uncap` part would already find the uncapitalized element in the map, so not add anything, but the element would not already be added in `files`, so it would be added there. The new implementation uses membership in `files` to decide whether to overwrite or not in both tables, which is different.


Consider for example adding `foo/A` and then `bar/a`; I would assume that the previous implement would end up with `files = {"A" => "foo/A", "a" => "bar/a"}, files_uncap = {"a" => "foo/A"}`, while the current implementation would end up with `files = {"A" => "foo/A", "a" => "bar/a"}, files_uncap = {"a" => "bar/a"}`.

Reply:
Good point. I've changed it to match the earlier behaviour.